solutions/usaco-640.mdx

```diff
@@ -137,13 +121,102 @@ int main()
 	}
 }
 
-//check if a piece is in bounds and is '#'
-bool check(int piece, int x, int y)
-{
+// check if a piece is in bounds and is '#'
+bool check(int piece, int x, int y) {
 	return x >= 0 && x < n && y >= 0 && y < n && pieces[piece][x][y];
 }
 ```
 
 </CPPSection>
 
+<JavaSection>
+```java
+import java.util.*;
+import java.io.*;
+
+public class BCS {
+	public static boolean pieces[][][];
+	public static int n;
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("bcs");
+	
+		pieces = new boolean[11][8][8];
+		int[][] s = new int[11][4];
```

Comment:
was thinking about this
how about you put:
11 in a variable called PIECE_NUM
dynamically initialize 8 with `n` instead (so `new boolean[PIECE_NUM][n][n]`)
and just do `new int[PIECE_NUM][]` instead of `new int[PIECE_NUM][4]`